lib/matplotlib/tests/test_mlab.py

```diff
@@ -25,91 +24,74 @@
+    assert_allclose(pca.Y[:, 2:], 0., atol=1e-8)
 
-    def test_prctile(self):
-        # test odd lengths
-        x = [1, 2, 3]
-        assert mlab.prctile(x, 50) == np.median(x)
 
-        # test even lengths
-        x = [1, 2, 3, 4]
-        assert mlab.prctile(x, 50) == np.median(x)
+def test_prctile():
+    # test odd lengths
+    x = [1, 2, 3]
+    assert mlab.prctile(x, 50) == np.median(x)
 
-        # derived from email sent by jason-sage to MPL-user on 20090914
-        ob1 = [1, 1, 2, 2, 1, 2, 4, 3, 2, 2, 2, 3,
-               4, 5, 6, 7, 8, 9, 7, 6, 4, 5, 5]
-        p = [0, 75, 100]
-        expected = [1, 5.5, 9]
+    # test even lengths
+    x = [1, 2, 3, 4]
+    assert mlab.prctile(x, 50) == np.median(x)
 
-        # test vectorized
-        actual = mlab.prctile(ob1, p)
-        assert_allclose(expected, actual)
+    # derived from email sent by jason-sage to MPL-user on 20090914
+    ob1 = [1, 1, 2, 2, 1, 2, 4, 3, 2, 2, 2, 3,
+           4, 5, 6, 7, 8, 9, 7, 6, 4, 5, 5]
+    p = [0, 75, 100]
+    expected = [1, 5.5, 9]
 
-        # test scalar
-        for pi, expectedi in zip(p, expected):
-            actuali = mlab.prctile(ob1, pi)
-            assert_allclose(expectedi, actuali)
+    # test vectorized
+    actual = mlab.prctile(ob1, p)
+    assert_allclose(expected, actual)
 
-    def test_norm(self):
-        np.random.seed(0)
-        N = 1000
-        x = np.random.standard_normal(N)
-        targ = np.linalg.norm(x)
-        res = mlab._norm(x)
-        assert_almost_equal(targ, res)
-
-
-class Test_spacing(CleanupTestCase):
-    def test_logspace_tens(self):
-        xmin = .01
-        xmax = 1000.
-        N = 6
-        res = mlab.logspace(xmin, xmax, N)
-        targ = np.logspace(np.log10(xmin), np.log10(xmax), N)
-        assert_allclose(targ, res)
-
-    def test_logspace_primes(self):
-        xmin = .03
-        xmax = 1313.
-        N = 7
-        res = mlab.logspace(xmin, xmax, N)
-        targ = np.logspace(np.log10(xmin), np.log10(xmax), N)
-        assert_allclose(targ, res)
-
-    def test_logspace_none(self):
-        xmin = .03
-        xmax = 1313.
-        N = 0
-        res = mlab.logspace(xmin, xmax, N)
-        targ = np.logspace(np.log10(xmin), np.log10(xmax), N)
-        assert_array_equal(targ, res)
-        assert res.size == 0
-
-    def test_logspace_single(self):
-        xmin = .03
-        xmax = 1313.
-        N = 1
-        res = mlab.logspace(xmin, xmax, N)
-        targ = np.logspace(np.log10(xmin), np.log10(xmax), N)
-        assert_array_equal(targ, res)
-        assert res.size == 1
-
-
-class Test_stride(CleanupTestCase):
+    # test scalar
+    for pi, expectedi in zip(p, expected):
+        actuali = mlab.prctile(ob1, pi)
+        assert_allclose(expectedi, actuali)
+
+
+def test_norm():
+    np.random.seed(0)
+    N = 1000
+    x = np.random.standard_normal(N)
+    targ = np.linalg.norm(x)
+    res = mlab._norm(x)
+    assert_almost_equal(targ, res)
```

Comment:
I wouldn't be opposed to it, but I'm not going to be the one to figure out which ones match those requirements. ;)